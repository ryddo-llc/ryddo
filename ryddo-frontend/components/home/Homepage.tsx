import React from 'react';
import HeroSection from './HeroSection';
import CategoryShowcase from './CategoryShowcase';
import PopularProducts from './PopularProducts';
import PaymentOptions from './PaymentOptions';

export default function Homepage() {
  return (
    <div className='w-full'>
      <HeroSection />
      <CategoryShowcase />
      <PaymentOptions />
      <PopularProducts />
    </div>
  );
}
