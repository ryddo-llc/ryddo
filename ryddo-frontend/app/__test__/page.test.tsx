import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Page from '../page';
import Homepage from '../../components/home/Homepage';

// Mock the Homepage component
jest.mock('../../components/home/Homepage', () => {
  return function MockHomepage() {
    return <div data-testid='mock-homepage'>Mocked Homepage</div>;
  };
});

describe('Page', () => {
  it('renders the Homepage component', () => {
    render(<Page />);
    expect(screen.getByTestId('mock-homepage')).toBeInTheDocument();
  });
});
