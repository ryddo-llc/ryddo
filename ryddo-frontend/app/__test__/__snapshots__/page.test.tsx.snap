// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders homepage unchanged 1`] = `
<div>
  <div
    data-testid="mock-homepage"
  >
    Mocked Homepage
  </div>
</div>
`;
